### ISSUE TITLE
Set public_header_files?

### ISSUE THREAD

**Issue by jberkel** [clemensg/sqlite3pod#8]
Right now, CocoaPods generates the following umbrella header file for sqlite3:

```c
#ifdef __OBJC__
#import <UIKit/UIKit.h>
#else
#ifndef FOUNDATION_EXPORT
#if defined(__cplusplus)
#define FOUNDATION_EXPORT extern "C"
#else
#define FOUNDATION_EXPORT extern
#endif
#endif
#endif

#import "sqlite3.h"
#import "sqlite3ext.h"

FOUNDATION_EXPORT double sqlite3VersionNumber;
FOUNDATION_EXPORT const unsigned char sqlite3VersionString[];
```

It includes both `sqlite3.h` and `sqlite3ext.h`. According to the [docs](https://sqlite.org/loadext.html#programming_loadable_extensions) the latter is only needed when building a loadable extension.

It redefines a bunch of macros to point to a global `sqlite3_api` struct. When including the sqlite3 pod using `@import sqlite3;` from an ObjC file, compilation fails with "use of undeclared identifier 'sqlite3_api'". The workaround is to set the compile flag `-DSQLITE_CORE` which basically skips `sqlite3ext.h`.

So I'm wondering if the real fix would be to explicitly set `s.public_header_files = "#{archive_name}/sqlite3.h"` (or similar) in the sqlite3 podspec to avoid the extension header, which is probably not needed in most cases and only causes problems.

**Comment by clemensg** [comment]
Hi, thanks for your report.

I will have a look at this. I also think that we don't really need the sqlite3ext.h header.

Setting `s.public_header_files` sounds like a good idea because SQLITE_CORE is probably for internal use only / for extension developers and is not documented as an official compile flag.

**Comment by clemensg** [comment]
Hmm. If I set `public_header_files` as you suggested, the specified header file is added to the existing ones and the umbrella header looks like this:
```
#import "sqlite3.h"
#import "sqlite3.h"
#import "sqlite3ext.h"
```

On the other hand, the documentation says that `public_header_files` is the way to avoid making all headers public. I am unsure why this is not working.

Maybe we need to create a module map?

If you find a solution, feel free to send a PR!

**Comment by jberkel** [comment]
Sure, I'll try my luck.

**Comment by clemensg** [comment]
Fixed by #9
